app/src/components/Sidebar.tsx

```diff
@@ -21,35 +21,62 @@ class Sidebar extends EnhancedComponent<ISidebarProps, ISidebarState> {
         {genre: "Rap", icon: "https://img.icons8.com/ios-glyphs/30/000000/country-music.png"},
     ]
 
+    private shownGenres: ISidebarGenreChannel[] = [];
```

Comment:
you don't really need this object here if it's just for the search algorithm, you can do the search in place with the state variable